### PR TITLE
ref(ts migration): replaced all instances of any in routeHandler.ts with explicit types

### PR DESCRIPTION
**Pull request by QiluXie**
## Problem

Closes [IS-834](https://isomeropengov.atlassian.net/browse/IS-834?atlOrigin=eyJpIjoiYzg1MGM2Mjk0ZGM1NGRkYzlmOTk3OTAxMWJjMWM4NTkiLCJwIjoiaiJ9)

There are many usages of `any` in the file `routeHandler.ts` which is not type safety in Typescript. 

## Solution

Replaced all usage of `any` with explicit types in `routeHandler.ts`

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [x] No - this PR is backwards compatible with ALL of the following feature flags in this [doc](https://www.notion.so/opengov/Existing-feature-flags-518ad2cdc325420893a105e88c432be5)

**Improvements**:

- Replaced all usage of `any` with explicit types in `routeHandler.ts`

## Tests

Run the code locally to confirm no type checking error or waiting for the CI to pass 

[IS-834]: https://isomeropengov.atlassian.net/browse/IS-834?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by seaerchin — COMMENTED** [review]
I think there's some confusion here regarding generics/specifying the type. 

we just want to give a concrete type to be safe here, we don't need the method to be generic as we want to narrow the types available.

**Comment by QiluXie** [comment]
> I think there's some confusion here regarding generics/specifying the type.
> 
> we just want to give a concrete type to be safe here, we don't need the method to be generic as we want to narrow the types available.

@seaerchin I actually tried this but got tons of type mismatching errors since our callers of the wrapper function are using very different concrete types. That's why I used generic types to let it infer from the callers, so we can avoid using `any`.

File: src/middleware/routeHandler.ts [hunk 1]
```diff
@@ -70,21 +73,37 @@ const isGitFileAndIsGitAvail = async (
...
+  Locals extends Record<string, unknown>
+>(
+  routeHandler: RequestHandler<P, ResBody, ReqBody, ReqQuery, Locals>
+) => async (
+  req: Request<P, ResBody, ReqBody, ReqQuery, Locals>,
+  res: Response<ResBody, Locals>,
+  next: NextFunction
 ) => {
-  routeHandler(req, res).catch((err: any) => {
+  Promise.resolve(routeHandler(req, res, next)).catch((err: Error) => {
```
[the remark this answers is not in the document]
Comment (QiluXie) [reply src/middleware/routeHandler.ts:89]: As discussed, the generics type is required here as the caller of the wrapper functions are deciding the concrete types which vary a lot. And it's the intended behaviour we want.

File: src/middleware/routeHandler.ts [hunk 2]
```diff
@@ -70,21 +73,37 @@ const isGitFileAndIsGitAvail = async (
...
+  ReqQuery,
+  Locals extends Record<string, unknown>
+>(
+  routeHandler: RequestHandler<P, ResBody, ReqBody, ReqQuery, Locals>
+) => async (
+  req: Request<P, ResBody, ReqBody, ReqQuery, Locals>,
+  res: Response<ResBody, Locals>,
+  next: NextFunction
 ) => {
   const { siteName } = req.params
```
[the remark this answers is not in the document]
Comment (QiluXie) [reply src/middleware/routeHandler.ts:108]: Similarly here, we need to use generic types and the global `Request` type already has `growthbook`

File: src/middleware/routeHandler.ts [hunk 3]
```diff
@@ -117,17 +136,33 @@ export const attachWriteRouteHandlerWrapper = (routeHandler: any) => async (
...
+  Locals extends {
+    userWithSiteSessionData: UserWithSiteSessionData
+    githubSessionData: GithubSessionData
+  }
+>(
+  routeHandler: RequestHandler<P, ResBody, ReqBody, ReqQuery, Locals>
+) => async (
+  req: Request<P, ResBody, ReqBody, ReqQuery, Locals>,
+  res: Response<ResBody, Locals>,
+  next: NextFunction
```
[the remark this answers is not in the document]
Comment (QiluXie) [reply src/middleware/routeHandler.ts:153]: Same here, generic types are needed.

File: src/middleware/routeHandler.ts [hunk 4]
```diff
@@ -35,7 +38,7 @@ const gitFileSystemService = new GitFileSystemService(simpleGitInstance)
 
 const handleGitFileLock = async (
   repoName: string,
-  next: (arg0: any) => void
+  next: (arg0: Error | null) => void
```
[the remark this answers is not in the document]
Comment (QiluXie) [reply src/middleware/routeHandler.ts]: Sure, changed it in c4211f18c4945c5e51a12d3c14e70630b188ff6d

File: src/middleware/routeHandler.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (seaerchin) [reply src/middleware/routeHandler.ts:83]: updated - we want to extract the wrapper function as a type so enforce guarantees across

**Review by seaerchin — APPROVED** [review]
lgtm

**Pull request merged**